spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/prometheus/PrometheusMetricsExportAutoConfiguration.java

```diff
@@ -117,13 +120,22 @@ public PrometheusPushGatewayManager prometheusPushGatewayManager(
 				PrometheusProperties prometheusProperties, Environment environment) {
 			PrometheusProperties.Pushgateway properties = prometheusProperties
 					.getPushgateway();
-			PushGateway pushGateway = new PushGateway(properties.getBaseUrl());
 			Duration pushRate = properties.getPushRate();
 			String job = getJob(properties, environment);
 			Map<String, String> groupingKey = properties.getGroupingKey();
 			ShutdownOperation shutdownOperation = properties.getShutdownOperation();
-			return new PrometheusPushGatewayManager(pushGateway, collectorRegistry,
-					pushRate, job, groupingKey, shutdownOperation);
+			return new PrometheusPushGatewayManager(
+					getPushGateway(properties.getBaseUrl()), collectorRegistry, pushRate,
+					job, groupingKey, shutdownOperation);
+		}
+
+		private PushGateway getPushGateway(String url) {
+			try {
+				return new PushGateway(new URL(Objects.toString(url, "")));
```

Comment:
I don't think we need to do `Objects.toString(url, "")` since the property has a default value of `localhost:9091`.